### PR TITLE
Link to the SonarQube dashboard on Gradle if SQ integration is enabled.

### PR DESCRIPTION
**Pull request by HSAR**

### CODE REVIEW

File: Tasks/Maven/CodeAnalysis/mavensonar.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ import sqCommon = require('sonarqube-common/sonarqube-common');
 // Gets the SonarQube tool runner if SonarQube analysis is enabled.
 export function getSonarQubeRunner(mvnPath:string, mavenPOMFile: string, mavenOptions: string, execFileJacoco?: string): ToolRunner {
     if (!sqCommon.isSonarQubeAnalysisEnabled()) {
-        return;
+        return null;
```
Comment (HSAR) [inline Tasks/Maven/CodeAnalysis/mavensonar.ts]: Unrelated change bundled into this PR - correctly returns null instead of just returning. This should never actually be hit as we check that SQ is enabled before calling into this method.

File: Tasks/Gradle/CodeAnalysis/gradlesonar.ts [hunk 2]
```diff
@@ -58,14 +53,19 @@ export function applyEnabledSonarQubeArguments(gradleRun: trm.ToolRunner):trm.To
 
 // Points SonarQube to the CC file as it is in a non-standard location. Not required for Jacoco. 
 export function applySonarQubeCodeCoverageArguments(gradleRun: trm.ToolRunner, isCodeCoverageEnabled:boolean, ccTool:string, reportPath:string):trm.ToolRunner {
-    
-    if (isSonarQubeEnabled() && isCodeCoverageEnabled && ccTool.toLowerCase() == "cobertura" && reportPath ) {
+    if (!sqCommon.isSonarQubeAnalysisEnabled()) {
```
Comment (bgavrilMS) [inline Tasks/Gradle/CodeAnalysis/gradlesonar.ts]: Why did you break up the if clause into 2 clauses?

File: Tasks/Gradle/gradletask.ts [hunk 3]
```diff
@@ -32,6 +37,7 @@ var testResultsFiles = tl.getInput('testResultsFiles', true);
 var summaryFile: string = null;
 var reportDirectory: string = null;
 var inputTasks: string[] = tl.getDelimitedInput('tasks', ' ', true);
+var isSonarQubeEnabled: boolean = false;
```
Comment (bgavrilMS) [inline Tasks/Gradle/gradletask.ts]: you can initialize this variable right here with its actual value

**Comment by bgavrilMS** [comment]
LGTM

File: Tasks/Gradle/CodeAnalysis/gradlesonar.ts (hunk 2 above)
Comment (HSAR) [inline Tasks/Gradle/CodeAnalysis/gradlesonar.ts]: This wasn't meant to be in the commit. However, this brings up the idea that perhaps we should move all the boolean checks up here in the return-early pattern.

File: Tasks/Gradle/gradletask.ts (hunk 3 above)
Comment (HSAR) [inline Tasks/Gradle/gradletask.ts]: Accepted.

Comment (HSAR) [inline Tasks/Gradle/gradletask.ts]: Resolved, done.

File: Tasks/Gradle/CodeAnalysis/gradlesonar.ts (hunk 2 above)
Comment (HSAR) [inline Tasks/Gradle/CodeAnalysis/gradlesonar.ts]: Accepted for change: re-combine the if statements.

Comment (HSAR) [inline Tasks/Gradle/CodeAnalysis/gradlesonar.ts]: Resolved, done.
